src/query/storage/cache/redis_cache.go

```diff
@@ -98,10 +98,17 @@ type CacheMetrics struct {
 
 	checkMismatchCounter tally.Counter
 	checkTotalCounter    tally.Counter
+
+	cacheBytesRetrieved tally.Histogram
+	m3dbBytesRetrieved  tally.Histogram
 }
 
 func NewCacheMetrics(scope tally.Scope) CacheMetrics {
 	subScope := scope.SubScope("cache")
+	// Histogram buckets range from 10^3 to 10^12 (scaled by a factor of 10 each time)
+	// Bucketting chosen to have enough buckets and a wide enough range for
+	// histogram_quantile() to be reliable
+	buckets, _ := tally.ExponentialValueBuckets(1000, 10, 10)
```

Comment:
does this include p99/p90 and p50 and p10?

Reply:
I can use the histogram_quantile() prometheus function to specify any quantile I think